testframes/test_mne_project_to_surface/test_mne_project_to_surface.cpp

```diff
@@ -0,0 +1,139 @@
+// INCLUDES
+//=============================================================================================================
+
+#include <utils/generics/applicationlogger.h>
+#include <mne/mne_project_to_surface.h>
+#include <mne/mne_bem.h>
+#include <mne/mne_bem_surface.h>
+
+//=============================================================================================================
+// QT INCLUDES
+//=============================================================================================================
+
+#include <QtCore/QCoreApplication>
+#include <QtTest>
+
+//=============================================================================================================
+// Eigen
+//=============================================================================================================
+
+#include <Eigen/Dense>
+
+//=============================================================================================================
+// Using NAMESPACE
+//=============================================================================================================
+
+using namespace MNELIB;
+using namespace Eigen;
+
+
+//=============================================================================================================
+/**
+ * DECLARE CLASS TestMNEProjectToSurface
+ *
+ * @brief The TestMNEProjectToSurface class provides read write read fiff verification tests
+ *
+ */
+class TestMNEProjectToSurface: public QObject
+{
+    Q_OBJECT
+
+public:
+    TestMNEProjectToSurface();
+
+private slots:
+    void initTestCase();
+    void compareValue();
+    // add other compareFunctions here
+    void cleanupTestCase();
+
+private:
+    // declare your thresholds, variables and error values here
+    double dEpsilon;
+    MatrixXf matResult;
+    MatrixXf matRef;
+
+};
+
+//=============================================================================================================
+
+TestMNEProjectToSurface::TestMNEProjectToSurface()
+    : dEpsilon(0.000001)
+{
+}
+
+//=============================================================================================================
+
+void TestMNEProjectToSurface::initTestCase()
+{
+    QFile t_fileBem(QCoreApplication::applicationDirPath() + "/mne-cpp-test-data/subjects/sample/bem/sample-1280-1280-1280-bem.fif");
+    MNEBem bemHead(t_fileBem);
+    MNEBemSurface::SPtr bemSurface = MNEBemSurface::SPtr::create(bemHead[0]);
+    MNEProjectToSurface::SPtr mneSurfacePoints = MNEProjectToSurface::SPtr::create(*bemSurface);
+
+    VectorXi vecNearest;                        // Triangle of the new point
+    VectorXf vecDist;                           // The Distance between matX and matP
+
+    matRef = bemSurface->tri_cent.cast<float>();
+    MatrixXf matTriCent = matRef * 1.5;         // Move all points with same amout from surface
+    int iNP = matTriCent.rows();
+    mneSurfacePoints->mne_find_closest_on_surface(matTriCent, iNP, matResult, vecNearest, vecDist);
```

Comment:
Do we need to give `matResult` a size before we pass it to `mne_find_closest_on_surface`? It seems that `vecNearest` and `vecDist` are resized inside `mne_find_closest_on_surface` but `matResult` is not.

Reply:
This was the reason that it failed on Mac and Linux tests. There was no resizing implemented. I can implement this, should not be a problem.

What is a problem is following: 
The idea does not work. I think the reason might be that from the bem I'm getting the triangle centers, but the function only projects to the edges on the trianglrs? Does this makes sense?

Reply:
And the surface is not a sphere. This might be the actual issue.